datasource.go

```diff
@@ -97,6 +88,33 @@ func (c *Client) UpdateDataSource(s *DataSource) error {
 	return nil
 }
 
+func (c *Client) DataSources() ([]DataSource, error) {
```

Comment:
It would be nice to have a test for this method, and even nicer that it'd have an example of the payload the api returns.

Reply:
We will prepare some additional tests.

Regarding payload, for example we use Azure Monitor and Zabbix, both having different jsonData and secureJsopData

Azure Monitori:
```
{
    "id": 101,
    "orgId": 1,
    "name": "SOME_DATASOURCE_NAME",
    "type": "grafana-azure-monitor-datasource",
    "typeLogoUrl": "",
    "access": "proxy",
    "url": "/api/datasources/proxy/101",
    "password": "",
    "user": "",
    "database": "",
    "basicAuth": false,
    "basicAuthUser": "",
    "basicAuthPassword": "",
    "withCredentials": false,
    "isDefault": false,
    "jsonData": {
        "appInsightsAppId": "00000000-0000-0000-0000-000000000000",
        "azureLogAnalyticsSameAs": false,
        "cloudName": "azuremonitor"
    },
    "secureJsonData": {
        "appInsightsApiKey":"bUtGHDSq7XtxsKeBUrfwYfk6QhGWZRP2
    },
    "version": 5,
    "readOnly": false
}
```

Zabbix:
```
{
    "id": 67,
    "orgId": 1,
    "name": "Zabbix",
    "type": "alexanderzobnin-zabbix-datasource",
    "typeLogoUrl": "",
    "access": "proxy",
    "url": "http://zabbi/api_jsonrpc.php",
    "password": "",
    "user": "",
    "database": "",
    "basicAuth": false,
    "basicAuthUser": "",
    "basicAuthPassword": "",
    "withCredentials": false,
    "isDefault": false,
    "jsonData": {
        "addThresholds": true,
        "alerting": true,
        "alertingMinSeverity": 3,
        "dbConnectionDatasourceId": null,
        "dbConnectionEnable": false,
        "disableReadOnlyUsersAck": false,
        "keepCookies": [],
        "password": "KdrVK8U0x3hgBppY",
        "trends": false,
        "username": "grafana",
        "zabbixVersion": 4
    },
    "secureJsonData": {},
    "version": 8,
    "readOnly": false
}
```